.github/composites/docker-build-push/action.yaml

```diff
@@ -0,0 +1,129 @@
+name: 'Build and Publish Docker Image'
+description: 'Build and Publish Docker Image'
+inputs:
+  custodian_image_var:
+    description: 'env var name for custodian image'
+    required: true
+  dockerfile:
+    description: 'Dockerfile to build'
+    required: true
+  repository:
+    description: 'docker hub repository'
+    required: true
+  image:
+    description: 'image'
+    required: true
+  name:
+    description: 'image name'
+    required: true
+  push:
+    description: 'push image: true/false'
+    required: true
+    default: false
+  poetry_version:
+    description: "Poetry Version to use"
+    default: "1.1.14"
+  trivy_version:
+    description: "trivy version to use"
+    default: "0.5.4"
+runs:
+  using: "composite"
+  steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+    - name: Install Docker Build Env
+      # bin directory is in .dockerignore
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip click
+        mkdir -p bin
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ env.HUB_USER }}
+        password: ${{ env.HUB_TOKEN }}
+
+    - name: determine tags
+      shell: bash
+      id: calctags
+      run: |
+        python tools/dev/dockerpkg.py tags -t auto
+        echo "::set-output name=tags::$(python tools/dev/dockerpkg.py tags -t auto)"
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: |
+          ${{ inputs.repository }}/${{ inputs.image }}
+        tags: |
+          # this is a hack to get around the fact that passing in tags with a new line character doesnt work
+          # ensure that we dont have more than 5 tags at a time
+          ${{ steps.calctags.outputs.tag0 }}
```

Comment:
the array stuff here feels funky and unused params feel like a vector, you can output a structured json and have it parsed with the action syntax.
https://docs.github.com/en/actions/learn-github-actions/expressions#fromjson